src/nvim/eval.c

```diff
@@ -10929,6 +10929,89 @@ static void f_getwininfo(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   }
 }
 
+// Dummy timer callback. Used by f_wait().
+static void dummy_timer_due_cb(TimeWatcher *tw, void *data)
+{
+  if (!uv_timer_get_repeat(&tw->uv)) {
+    time_watcher_start(tw, dummy_timer_due_cb, 0, 0);
+  }
+}
+
+// Dummy timer close callback. Used by f_wait().
+static void dummy_timer_close_cb(TimeWatcher *tw, void *data)
+{
+  multiqueue_free(tw->events);
+  xfree(tw);
+}
+
+/// "wait(timeout, condition[, interval])" function
+static void f_wait(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  rettv->v_type = VAR_NUMBER;
+  rettv->vval.v_number = -1;
+
+  if (argvars[0].v_type != VAR_NUMBER) {
+    EMSG2(_(e_invarg2), "First argument of wait() must be a number");
+    return;
+  }
+
+  int timeout = argvars[0].vval.v_number;
+  typval_T expr = argvars[1];
+
+  int interval = -1;
+  typval_T *tv_interval = &argvars[2];
+
+  TimeWatcher *tw = NULL;
+
+  if (tv_interval->v_type == VAR_NUMBER) {
+    interval = tv_interval->vval.v_number;
+    if (interval < 0) {
+      EMSG2(_(e_invarg2),
+            "Third argument of wait() must be a non-negative number");
+      return;
+    }
+    // Start dummy timer
+    tw = xmalloc(sizeof(TimeWatcher));
+    time_watcher_init(&main_loop, tw, NULL);
+    tw->events = multiqueue_new_child(main_loop.events);
```

Comment:
Can just use `tw->events = main_loop.events` and remove `multiqueue_free(tw->events)`. (`blockable=true` is still ok, because if there is another another event already on the loop the condition will be evaluated anyway).